src/Collection/CollectionTrait.php

```diff
@@ -512,7 +512,7 @@ public function insert($path, $values)
     public function toArray($preserveKeys = true)
     {
         $iterator = $this->unwrap();
-        if ($iterator instanceof ArrayIterator) {
+        if (get_class($iterator) === ArrayIterator::class) {
```

Comment:
I think this case is safe to keep as it was before. We are about to return the array values anyway